src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,87 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo | null;
+  setSelectedTodo: Dispatch<SetStateAction<Todo | null>>;
+  isLoadingModal: boolean;
+  setLoadingModal: Dispatch<SetStateAction<boolean>>;
```

Comment:
```suggestion
  selectedTodo: Todo;
  setSelectedTodo: Dispatch<SetStateAction<Todo | null>>;
```
- You can render modal only when you have selected todo
- keep loading state in the component